### PR TITLE
Added a tutorial for model loading based on the model loading tutorial from LearnOpenGL

### PR DESCRIPTION
**Pull request by zoeysaurusrex**
# Summary of the PR
- Adds a tutorial for model loading, based on the "Model Loading" tutorial from [LearnOpenGL](https://learnopengl.com/Model-Loading/Model)
- Based on the numbering scheme of the other tutorials against the existing LearnOpenGL set, I guessed that model loading would be 4.1, but I'm happy to change it to whatever the correct thing is :)

# Related issues, Discord discussions, or proposals
#621 

# Further Comments
- Still a WIP. Native/unmanaged interop isn't my strongest skill with C# and I'm open to any feedback in case I'm doing anything silly. 
- For the sake of this tutorial, the model file is uploaded as cube.model, but it is a .obj file. I'm expecting we may need to reconcile the gitignore rules for .obj files, or I could try a different model format.
- The tutorial project references the Assimp nuget package (set to any version) because referencing the Assimp or Assimp native projects didn't push Assimp*.dll to the bin folder and I felt I might not properly understand the intent of the native project. I'm happy to change it if there is a way to get that to work :)

### CODE REVIEW

**Comment by zoeysaurusrex** [comment]
Tagging maintainers in accordance with the readme :) 
@HurricanKai, @ThomasMiz, @Beyley

**Comment by Perksey** [comment]
Sorry for the delay @ThomasMiz do you have the capacity to take this one?

**Comment by ThomasMiz** [comment]
> Sorry for the delay @ThomasMiz do you have the capacity to take this one?

I was planning on getting to it before the community meeting, I've just been very busy these past few months and got burnt out

File: examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Model.cs
Comment (ThomasMiz) [inline examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Model.cs:14]:
```suggestion
using Silk.NET.Assimp;
using Silk.NET.OpenGL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using AssimpMesh = Silk.NET.Assimp.Mesh;
```

Is it just me getting that these imports are not used?

File: examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Program.cs
Comment (ThomasMiz) [inline examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Program.cs:99]:
```suggestion
            Gl.Clear(ClearBufferMask.ColorBufferBit | ClearBufferMask.DepthBufferBit);
```

People usually prefer the enum overloads

File: examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Program.cs
Comment (ThomasMiz) [inline examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Program.cs:132]:
```suggestion
            if (LastMousePosition == default)
            {
                LastMousePosition = position;
            }
            else
```

File: examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Model.cs
Comment (ThomasMiz) [inline examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Model.cs:51]:
```suggestion
        private unsafe void LoadModel(string path)
        {
            var scene = _assimp.ImportFile(path, (uint) PostProcessSteps.Triangulate);
            if (scene == null || scene->MFlags == Silk.NET.Assimp.Assimp.SceneFlagsIncomplete || scene->MRootNode == null)
            {
                var error = _assimp.GetErrorStringS();
                throw new Exception(error);
            }
            Directory = path;
            ProcessNode(scene->MRootNode, scene);
        }
```

File: examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Model.cs
Comment (ThomasMiz) [inline examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Model.cs:68]: Good job with the explanatory comments in this method!

File: examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Program.cs
Comment (ThomasMiz) [inline examples/CSharp/OpenGL Tutorials/Tutorial 4.1 - Model Loading/Program.cs:124]:
```suggestion
                Gl.DrawArrays(PrimitiveType.Triangles, 0, (uint)mesh.Vertices.Length);
```

**Comment by ThomasMiz** [comment]
Good job with this one and thanks for the contribution!

The tutorial ran fine first time without any issues and the code is a small sample on how to use Assimp, just like we want. I've never used Assimp myself but this little demo is clear enough for me and others know how to get started with it ;)

I've left a few suggestions for consistent code style

**Comment by Perksey** [comment]
I've applied your fixes @ThomasMiz as this PR was at risk of going stale. Provided build... builds... this should be good to go.

**Review by Perksey — APPROVED** [review]

**Comment by tilkinsc** [comment]
Doesn't assimp emit indices? It would be ideal to include that in here.

**Comment by Perksey** [comment]
Yeah that’d be a good idea. However I’m keen to get at least something in the repo, so once this is merged we’ll happily take a contribution improving it to use indices (iirc Assimp optimises or can be configured to optimise the vertices so that there are no duplicate verts, so definitely a lot of gain to be had with indices)

**Comment by tilkinsc** [comment]
Duplicate verts are ideally handled by the editor, as the duplicated vertices can better be handled by distance rather than obscure postprocessing by the model loading for each instance of the client lol. But yeah now thats its merged I'll likely contribute the code once I get assimp installed into my engine. It's been really difficult setting up with the IInputContext and IWindow due to how you have to lop everything into the load event for initialization.

**Pull request merged**
